tmc-langs-framework/src/test/java/fi/helsinki/cs/tmc/langs/ClassPathTest.java

```diff
@@ -0,0 +1,74 @@
+
+package fi.helsinki.cs.tmc.langs;
+
+import com.google.common.base.Throwables;
+import java.io.File;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ClassPathTest {
+    
+    private ClassPath cp;
+    
+    @Before
+    public void setUp() {
+        cp = new ClassPath(getPath("ant_project"));
+    }
+    
+    @Test
+    public void classPathConstructorWorks() {
+        assertTrue("ClassPath didn\'t contain path that was passed in constructor", cp.toString().contains("ant_project"));
+        assertEquals("Wrong amount of subpaths ", 1, cp.getPaths().size());
+    }
+    
+    @Test
+    public void addingPathAddsPathToSubpaths() {
+        cp.add(getPath("arith_funcs"));
+        assertTrue("ClassPath didn\'t contain path that was added", cp.toString().contains("arith_funcs"));
+        assertEquals("Wrong amount of subpaths ", 2, cp.getPaths().size());
+    }
+    
+    @Test
+    public void addingSamePathDoesntAddItTwice() {
+        cp.add(getPath("arith_funcs"));
+        cp.add(getPath("arith_funcs"));
+        assertTrue("ClassPath didn\'t contain path that was added", cp.toString().contains("arith_funcs"));
+        assertEquals("Wrong amount of subpaths ", 2, cp.getPaths().size());
+    }
+    
+    @Test
+    public void addingAnotherClassPathAddsNewSubpaths() {
+        ClassPath classPath = new ClassPath(getPath("arith_funcs"));
+        assertFalse(cp.toString().contains("arith_funcs"));
+        cp.add(classPath);
+        assertTrue("ClassPath didn\'t contain path that was added", cp.toString().contains("arith_funcs"));
+        assertEquals("Wrong amount of subpaths ", 2, cp.getPaths().size());
+    }
+
+    @Test
+    public void addDirAndSubDirAddsDirAndSubDirsToPaths() {
+        cp.addDirAndContents(getPath("arith_funcs" + File.separatorChar + "lib"));
+        assertTrue("ClassPath didn\'t contain edu-test-utils dir", cp.toString().contains("edu-test-utils"));
```

Comment:
Was testing for one of the jars that should be added to the classpath, changed the test conditions.